enhancements/installer/vsphere-ipi-zonal-external-lb.md

```diff
@@ -0,0 +1,201 @@
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The goal of this enhancement is to provide the ability to install vSphere IPI zonal on separate L2 segments without the static pods (keepalived, haproxy, coredns). 
+As of this writing the only way to accomplish this is to allow an external load balancer.
+
+## Motivation
+
+Users of OpenShift would like the ability to deploy control plane and compute nodes in more than one L2 segment to increase reliability and ease of network management.
+
+### Goals
+
+- Support external load balancer in vSphere IPI zonal
+- Support multiple subnets via the external load balancer
+- Support for no static pod deployment (keepalived, haproxy, coredns)
+
+### Non-Goals
+
+- Provide a mechanism to configure a external load balancer from the installer or machine-api-operator 
+
+## Proposal
+
+We propose the use of small L2 segments (e.g. /28). The customer's
+load balancer will be pre-configured with the entire subnet as end points
+for the required ports. Those L2 segments will be a associated with vSphere port groups that will be used in the `FailureDomains` and provisioning process.  The external load balancer will discover available endpoints and use those accordingly.
+
+- Modification to the Infrastructure status object to include new LoadBalancerType
+- Modification to the platform spec object to include new LoadBalancerType
+- Modification to the machine-config-operator to use new LoadBalancerType 
+
+### Workflow Description
+
+### api
+
+#### Infrastructure Status
+
+```golang
+// LoadBalancerType can be either External or Internal 
+type LoadBalancerType string
+
+const (
+	// LoadBalancerTypeExternal
+	LoadBalancerTypeExternal LoadBalancerType = "External"
+
+	// LoadBalancerTypeInternal
+	LoadBalancerTypeInternal LoadBalancerType = "Internal"
+)
+
+// VSpherePlatformStatus holds the current status of the vSphere infrastructure provider.
+type VSpherePlatformStatus struct {
+  // loadBalancerType defines if a internal or external load balancer
+  // is to be used for the openshift api and ingress endpoints.
+  // When set to Internal the static pods defined in machine config operator
+  // will be used.
+  // When set to External the static pods will not be deployed. It will be 
+  // expected that the LB and DNS A records be pre-configured prior to installation.
+  // +kubebuilder:default=External
+  // +kubebuilder:validation:Enum=Internal;External
+  LoadBalancer LoadBalancerType `json:"loadBalancerType,omitempty"`
+}
+```
+
+### MCO
+
+The existing switch to deploy keepalived, haproxy and coredns static pods in the machine config operator is the apivip variable. This would change to be the Infrastructure status parameter: `LoadBalancerType`.
```

Comment:
for all on prem platforms or only for vsphere?
I would prefer to have it a generic enhancement to the on prem networking